### PR TITLE
Add  "doctrine/collections"

### PR DESCRIPTION
**Pull request by jmontoyaa**
According to this:

https://github.com/Sylius/Attribute/blob/master/Model/AttributeSubjectInterface.php#L14

Doctrine collections must be present ...

### CODE REVIEW

**Comment by okwinza** [comment]
IIRC there was talk about replacing it with something generic/sylius based.

**Comment by jmontoyaa** [comment]
In the mean time 0.16 was released with Collection :smile:

**Comment by pamil** [comment]
IMHO methods like `setCollection` should be removed as they are very rarely used and can be replaced through iterating and calling `addCollectionElement`. They also couple model interfaces with Doctrine Collections.

**Comment by pamil** [comment]
For example, this method is never used in Sylius codebase.

**Comment by pjedrzejewski** [comment]
:+1: for removing the setters, especially if we do not use them but that still leaves the deps here https://github.com/Sylius/Attribute/blob/master/Model/Attribute.php#L63.

**Comment by michalmarcinkowski** [comment]
Yeah, but removing `Collection` from the interface at least do not enforce you to use it in your implementation, so that's a huge benefit.

File: src/Sylius/Component/Attribute/composer.json [hunk 1]
```diff
@@ -25,7 +25,8 @@
         "sylius/resource": "0.17.*",
         "sylius/translation": "0.17.*",
         "sylius/registry": "0.17.*",
-        "symfony/validator": "^2.7.7"
+        "symfony/validator": "^2.7.7",
+        "doctrine/collections": "~1.2"
```
Comment (lchrusciel) [inline src/Sylius/Component/Attribute/composer.json]: Can we use lower version?

Comment (jmontoyaa) [inline src/Sylius/Component/Attribute/composer.json]: I got that value from here:

https://github.com/Sylius/Sylius/blob/master/composer.json#L28

Comment (lchrusciel) [inline src/Sylius/Component/Attribute/composer.json]: :+1:

**Comment by michalmarcinkowski** [comment]
Thanks Julio! :+1:

**Pull request merged**